src/components/FlowPlayer/FlowPlayer.tsx

```diff
@@ -150,14 +115,54 @@ export const FlowPlayer: FunctionComponent<FlowPlayerProps> = ({
 	}, [start, end]);
 
 	// Draw custom elements
-	flowplayer((opts: any, root: any, api: any) => {
-		const mq = flowplayer.mq;
+	useLayoutEffect(() => {
+		const createTitleOverlay = () => {
+			const titleOverlay = createElement('div', 'c-title-overlay');
+			const titleHeader = createElement('h5', 'c-title-overlay__title');
+			const publishDiv = createElement('div', 'u-d-flex');
+
+			titleHeader.innerText = title;
+
+			titleOverlay.appendChild(titleHeader);
+			titleOverlay.appendChild(publishDiv);
+
+			if (subtitles && subtitles.length) {
+				subtitles.forEach((subtitle: string) => {
+					const substitleDiv = createElement('div', 'c-title-overlay__meta');
+
+					substitleDiv.innerText = subtitle;
+					publishDiv.appendChild(substitleDiv);
+				});
+			}
+
+			return titleOverlay;
+		};
 
-		api.on('mount', () => {
-			mq('.fp-ui', root).prepend(createTitleOverlay());
-			mq('.fp-ui', root).prepend(createLogoOverlay());
+		const createLogoOverlay = () => {
+			if (logo) {
+				const logoOverlay = createElement('div', 'c-logo-overlay');
+				const logoImg = createElement('img', 'c-logo-overlay__img');
+
+				logoImg.src = logo;
+				logoOverlay.appendChild(logoImg);
+
+				return logoOverlay;
+			}
+		};
+
+		flowplayer((opts: any, root: any, api: any) => {
+			const mq = flowplayer.mq;
+
+			// Check if root equals current components container ref
+			// to avoid creating multiple titles and logos
+			if (root === videoContainerRef.current) {
```

Comment:
I don't think this solution is ideal, maybe we should give the FlowPlayer component a unique id and add that to the element for checking?